src/shim/kernel.rs

```diff
@@ -96,7 +71,21 @@ error_number! {
 
 impl fmt::Display for ErrorNumber {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        self.as_unshimmed().fmt(f)
+        match self {
+            Self::IllegalArgument => f.write_str("illegal argument"),
+            Self::IllegalOperation => f.write_str("illegal operation"),
+            Self::LimitExceeded => f.write_str("limit exceeded"),
+            Self::AssertionFailed => f.write_str("filecoin assertion failed"),
+            Self::InsufficientFunds => f.write_str("insufficient funds"),
+            Self::NotFound => f.write_str("resource not found"),
+            Self::InvalidHandle => f.write_str("invalid ipld block handle"),
+            Self::IllegalCid => f.write_str("illegal cid specification"),
+            Self::IllegalCodec => f.write_str("illegal ipld codec"),
+            Self::Serialization => f.write_str("serialization error"),
+            Self::Forbidden => f.write_str("operation forbidden"),
+            Self::BufferTooSmall => f.write_str("buffer too small"),
+            Self::Unknown(u) => write!(f, "{}", u),
```

Comment:
I don't like this - is it a pasted block from fvm?
It's not our responsibility to define these strings - can you think of a way to defer definitions to fvm?

Reply:
Yes, will try to find something better.

Reply:
@aatifsyed Is my new approach better?